### ISSUE TITLE
MapConfig doesn't work unless name is prefixed with "hz:"

### ISSUE THREAD

**Issue by kedar-joshi** [hazelcast/hazelcast#14932]
See sample code given below -

```java
// final String name = "hz:application-uuid-map"; // Works
final String name = "application-uuid-map"; // Doesn't work
final MapConfig mapConfig = new MapConfig(name).setEvictionPolicy(EvictionPolicy.LRU)
			.setTimeToLiveSeconds(3600);

final Config config = new Config().addMapConfig(mapConfig)
			.setProperty(GroupProperty.LOGGING_TYPE.getName(), "log4j2");
final HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(config);
final IMap<Object, Object> map = hazelcastInstance.getMap(name);

final Object o = map.get(1L);

hazelcastInstance.shutdown();
```

When config name is  **application-uuid-map**, following message is printed during application startup -

```
2019-04-25 11:31:01.637 [hz._hzInstance_1_dev.partition-operation.thread-6] TRACE com.hazelcast.config.Config : No configuration found for application-uuid-map, using default config!
2019-04-25 11:31:01.638 [hz._hzInstance_1_dev.partition-operation.thread-6] TRACE com.hazelcast.config.Config : No configuration found for application-uuid-map, using default config!
```
When prefixing config name with **hz:** (i.e. **hz:application-uuid-map**) this message is not printed.

> No configuration found for application-uuid-map, using default config!

The message suggests that user configuration is not considered and default one is used instead. Also, no example in Hazelcast documentation mentions such expectation.

**Environment**

Windows 10 x64
Hazelcast 3.12
OpenJDK 12.0.1

**Comment by ahmetmircik** [comment]
Thanks for pointing this. 

Log message is misleading, `default` is not your map's configuration, your map gets correct map-config for `application-uuid-map` but map-event-journal can't find an explicit config for itself and printing this log. This is harmless. But log message should be clear. 

I'll keep this issue open as an enhancement.

**Comment by mmedenjak** [comment]
Fixed in 4.0 with https://github.com/hazelcast/hazelcast/pull/15185. We no longer need to search for the journal configuration as there is a reference from MapConfig.
